src/Features/CSharp/Portable/CodeFixes/AddExplicitCast/AddExplicitCastCodeFixProvider.cs

```diff
@@ -0,0 +1,399 @@
+                        matchedTypes[parameterIndex] = true;
+                    }
+                    else return false;
+
+                    if (targetArgument.Equals(arguments[i])) targetParamIndex = parameterIndex;
+                }
+                else return false;
+            }
+
+            // mark all optional parameters as matched
+            for (var i = 0; i < parameters.Length; i++)
+            {
+                if (parameters[i].IsOptional || parameters[i].IsParams)
+                {
+                    matchedTypes[i] = true;
+                }
+            }
+
+            return Array.TrueForAll(matchedTypes, (item => item));
+        }
+
+        class InheritanceDistanceComparer : IComparer<ITypeSymbol>
+        {
+            private ITypeSymbol baseType;
+            private SemanticModel semanticModel;
+
+            private int GetInheritanceDistance(ITypeSymbol baseType, ITypeSymbol? derivedType)
+            {
+                if (derivedType == null) return int.MaxValue;
+                if (derivedType.Equals(baseType)) return 0;
+
+                var distance = int.MaxValue;
+                distance = Math.Min(GetInheritanceDistance(baseType, derivedType.BaseType), distance);
+
+                if (derivedType.Interfaces.Length != 0)
+                {
+                    foreach (var interface_type in derivedType.Interfaces)
+                    {
+                        distance = Math.Min(GetInheritanceDistance(baseType, interface_type), distance);
+                    }
+                }
+
+                return distance == int.MaxValue ? distance : distance + 1;
+            }
+            public int Compare(ITypeSymbol x, ITypeSymbol y)
+            {
+                // if the node has the explicit conversion operator, then it has the shortest distance
+                var x_dist = semanticModel.Compilation.ClassifyCommonConversion(baseType, x).IsUserDefined ?
+                    0 : GetInheritanceDistance(baseType, x);
+                var y_dist = semanticModel.Compilation.ClassifyCommonConversion(baseType, y).IsUserDefined ?
+                    0 : GetInheritanceDistance(baseType, y);
+                return x_dist.CompareTo(y_dist);
+            }
+
+            public InheritanceDistanceComparer(SemanticModel semanticModel, ITypeSymbol baseType)
+            {
+                this.semanticModel = semanticModel;
+                this.baseType = baseType;
+            }
+        }
+
+
+        protected SyntaxNode? TryGetTargetNode(SyntaxNode root, TextSpan span)
+        {
+            var ancestors = root.FindToken(span.Start).GetAncestors<SyntaxNode>();
+
+            var node = ancestors.FirstOrDefault(n => n.Span.Contains(span) && n != root);
+            return node;
+        }
+
+        protected bool TryGetNode(SyntaxNode root, TextSpan span, SyntaxKind kind, SyntaxNode target, out SyntaxNode? node)
```

Comment:
This method can also be inlined as `if (target.GetAncestors<ArgumentSyntax>().FirstOrDefault() is ArgumentSyntax argument`